src/Resources/JsonApiResourceCollection.php

```diff
@@ -8,18 +8,27 @@
 
 class JsonApiResourceCollection extends ResourceCollection
 {
+    /**
+     * @var Collection<int, JsonApiResource>
+     */
+    public $collection;
+
     /**
      * Build the response
      *
      * @param  Request  $request
      */
     public function toArray($request): array
     {
-        $response = [
-            'data' => $this->collection,
-        ];
+        if ($this->collection->first()->isRegistered()) {
+            $response = [
+                'data' => $this->collection,
+            ];
+
+            return $this->addToResponse($request, $response);
+        }
 
-        return $this->addToResponse($request, $response);
+        return parent::toArray($request);
     }
```

Comment:
Why is this method needed? Since everything is put though the 'toArray' method now, right?